nix/default.nix

```diff
@@ -34,6 +39,16 @@ let
       # This contains musl-specific stuff, but it's all guarded by appropriate host-platform
       # checks, so we can include it unconditionally
       (import ./overlays/musl.nix)
+      # Disable profiling libraries in ghc883 to keep the devcontainer size down
+      (final: prev: {
+        haskell-nix = prev.haskell-nix // {
+          compiler = prev.haskell-nix.compiler // {
+            ghc883 = prev.haskell-nix.compiler.ghc883.override {
+              inherit enableLibraryProfiling;
```

Comment:
This is specifically the libraries that ship with GHC? This is the only place we use this argument, so I guess this won't actually affect the profiling for other packages? I guess I'd prefer not to do this in Nix and control it via the `cabal.project` settings like usual. Or is there a reason we have to do it this way.

Reply:
> This is specifically the libraries that ship with GHC?

This turns it on or off for the libraries in ghc itself.   It is only because we are putting ghc in the devcontainer that we and that the size therefor matters more that we would turn it off.

> This is the only place we use this argument, so I guess this won't actually affect the profiling for other packages?

I don't think it will be possible to link profiled versions of other packages with disabled.

> I guess I'd prefer not to do this in Nix and control it via the cabal.project settings like usual. Or is there a reason we have to do it this way.

Not sure which cabal.project file we would put it in.  Also if profiling is not enabled in a `cabal.project` file you would not want haskell.nix to install a different ghc as a result.

Ideally we would split the profiled versions of ghc's libraries out into another output of the ghc derivation instead (but that looks like it might be tricky).  Then profiled builds could use a reference to that.

Reply:
I take the argument about devcontainer size, but at the moment I don't care *that* much about it, and I care more about the maintainability of our Nix expressions. How much of a difference does it make?

> Also if profiling is not enabled in a cabal.project file you would not want haskell.nix to install a different ghc as a result.

Good point. But equally that means if I do enable profiling it will... not work? Break during the derivation build?